### PR TITLE
Add MySQL 8 Support to Backup Tests

### PR DESCRIPTION
**Pull request by mattlord**
## Description

This PR does the following:
1. Modifies the MySQL 8.0 CI workflow template to use the Percona apt repo instead of the MySQL one
    - This then aligns the mysqld and xtrabackup versions
    - xtrabackup 8.0.28 is latest and does not support mysqld 8.0.29, which is the latest server version upstream
2. Adds explicit MySQL 8.0 and XtraBackup 8.0 support to the relevant endtoend backup tests
    - The tar stream specific tests are skipped as those are [not supported in XtraBackup 8.0](https://docs.percona.com/percona-xtrabackup/8.0/xtrabackup_bin/backup.streaming.html)
3. Creates a new `(xb_backup) mysql80` workflow to test XtraBackup 8.0 with MySQL 8.0
4. Renames the test shard/workflow "19" to the more meaningful "vtbackup_transform"
5. Modifies how the xtrabackup related tests run with 5.7 so that the only tests done (not skipped) [are the tar stream specific ones](https://github.com/vitessio/vitess/blob/main/go/test/endtoend/backup/xtrabackup/xtrabackup_test.go#L27-L30)
    - Meaning that the xbstream and vtctlbackup tests are only done with 8.0
    - This is because while [XtraBackup 2.4](https://docs.percona.com/percona-xtrabackup/2.4/innobackupex/streaming_backups_innobackupex.html) defaults to using the tar stream method, [XtraBackup 8.0](https://docs.percona.com/percona-xtrabackup/8.0/xtrabackup_bin/backup.streaming.html) does not have a tar stream method, only xbstream
    
## Checklist

-   [x] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

**Comment by rsajwani** [comment]
changes looks good. These are precisely all the issue I encounter on my local box. Thanks for fixing them.

**Review by rsajwani — APPROVED** [review]

**Review by deepthi — COMMENTED** [review]
Nice work @mattlord, thank you for taking this on.
I'm a little iffy about changing ALL 8.0 tests to use percona. I would prefer to do that only for the xtrabackup test clusters.

File: go/mysql/collations/tools/makecolldata/contractions.go [hunk 1]
```diff
@@ -58,7 +58,7 @@ func printContraction1(g *codegen.Generator, wa *weightarray, incont []uca.Contr
 			trie[r] = append(trie[r], cont)
 		}
 		if depth == len(cont.Path) {
-			leaf = &cont
+			leaf = &cont // nolint:exportloopref
```
[the remark this answers is not in the document]
Comment (mattlord) [reply go/mysql/collations/tools/makecolldata/contractions.go:61]: Thanks! I'll remove and rebase.

[the remark this answers is not in the document]
Comment (mattlord) [reply go/mysql/collations/tools/makecolldata/contractions.go:61]: > Nice work @mattlord, thank you for taking this on.
> I'm a little iffy about changing ALL 8.0 tests to use percona. I would prefer to do that only for the xtrabackup test clusters.

I don't see a problem with it. Percona 8.0 has little changes beyond upstream as most of the patches that were in 5.7 and before (especially 5.6 and 5.5) are now no longer relevant. I also like that we're testing what I think many of our users are using. But I understand the concern.

I can make this change in only the XtraBackup workflows though. I'll just need to create a separate template for them and handle them specially in the generator. If you feel strongly about it then I'll do that.

[the remark this answers is not in the document]
Comment (deepthi) [reply go/mysql/collations/tools/makecolldata/contractions.go:61]: In that case, this LGTM

**Review by deepthi — APPROVED** [review]

File: go/mysql/collations/tools/makecolldata/contractions.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mattlord) [reply go/mysql/collations/tools/makecolldata/contractions.go:61]: Here's a patch we can leverage if we do decide to change this decision in the future and limit Percona Server 8.0 usage to XtraBackup tests: https://gist.github.com/mattlord/341fc54e4fa3f36d11bd03d8bdf57124

**Pull request merged**